frontend/src/pages/UserEditPage.jsx

```diff
@@ -1,14 +1,299 @@
+            type='text'
+            id='username'
+            name='username'
+            value={formData.username}
+            onChange={handleChange}
+            className='input input-bordered w-full p-2 rounded-md'
+            disabled={isUpdating}
+          />
+          {formErrors.username && (
+            <p className='text-error text-sm mt-1'>{formErrors.username}</p>
+          )}
+        </div>
+
+        <div className='mb-4'>
+          <label
+            htmlFor='name'
+            className='block text-lg font-medium text-gray-700 mb-2'
+          >
+            Name:
+          </label>
+          <input
+            type='text'
+            id='name'
+            name='name'
+            value={formData.name}
+            onChange={handleChange}
+            className='input input-bordered w-full p-2 rounded-md'
+            disabled={isUpdating}
+          />
+          {formErrors.name && (
+            <p className='text-error text-sm mt-1'>{formErrors.name}</p>
+          )}
+        </div>
+
+        {isAdmin && (
+          <div className='mb-4'>
+            <label className='block text-lg font-medium text-gray-700 mb-2'>
+              Admin Status:
+            </label>
+            <div className='flex items-center space-x-4'>
+              <label className='inline-flex items-center'>
+                <input
+                  type='radio'
+                  name='is_admin'
+                  value='true'
+                  checked={formData.is_admin === true}
+                  onChange={handleChange}
+                  className='radio radio-primary'
+                  disabled={isUpdating}
+                />
+                <span className='ml-2 text-gray-700'>Yes</span>
+              </label>
+              <label className='inline-flex items-center'>
+                <input
+                  type='radio'
+                  name='is_admin'
+                  value='false'
+                  checked={formData.is_admin === false}
+                  onChange={handleChange}
+                  className='radio radio-primary'
+                  disabled={isUpdating}
+                />
+                <span className='ml-2 text-gray-700'>No</span>
+              </label>
+            </div>
+          </div>
+        )}
+
+        <div className='flex justify-center gap-4 mt-6'>
+          <button
+            type='button'
+            onClick={() => navigate(`/users/${id}`)}
+            className='animated-button !text-base !px-5 !py-2 !bg-gray-500 hover:!bg-gray-600'
+            disabled={isUpdating}
+          >
+            Cancel
+          </button>
+          <button
+            type='submit'
+            className='animated-button !text-base !px-5 !py-2'
+            disabled={isUpdating}
+          >
+            {isUpdating ? 'Saving...' : 'Save Changes'}
+          </button>
+        </div>
+      </form>
+
+      <dialog ref={confirmModalRef} className='modal' open={showConfirmModal}>
```

Comment:
I'm not sure if confirmation modal is needed here. It will additionally cover the inputs, what makes asking about confirmation rely on the user's memory.